src/main/scala/autoversion/model/ConventionalCommit.scala

```diff
@@ -0,0 +1,32 @@
+package autoversion.model
+
+case class ConventionalCommit(
+  commitMessage: String,
+  kind: String,
+  breaking: Boolean = false,
+  scope: Option[String] = None,
+  description: Option[String] = None
+)
+
+object ConventionalCommit {
+
+  def parse(message: String): Option[ConventionalCommit] = {
+    val pattern = """^(?<kind>[\w\s]+)\s*(\((?<scope>[\w\s]+)\))?\s*(?<breaking>!)?\s*:\s+(?<description>.+)$"""
+    val regex = pattern.r("kind", "outerscope", "scope", "breaking", "description")
+
+    regex.findFirstMatchIn(message).flatMap { m =>
+      val breaking = Option(m.group("breaking")).isDefined || message.contains("BREAKING CHANGE")
+
+      for {
+        kind <- Option(m.group("kind"))
+      } yield ConventionalCommit(
```

Comment:
Very nitpicky, but with a single Option, I'd go with a flatMap rather than a for comprehension ;)